infrastructure/smart-contracts/test/utils/submitMockRollupChallenge.mjs

```diff
@@ -0,0 +1,88 @@
+/**
+ * Submits a test challenge to the mock rollup contract.
+ * NOTE: the toggleAssertionChecking() and setRollupAddress() function must exist on the referee
+ * contract for this helper function to operate.
+ * @param {Contract} referee - The referee contract instance to submit the challenge to.
+ * @param {Wallet} challengerWallet - The wallet of the challenger submitting the challenge.
+ * @param {Contract} mockRollup - The MockRollup contract instance.
+ * @param {Contract} refereeCalculations - The RefereeCalculations contract instance.
+ * @param {BigInt} currentAssertion - The current assertion id for the challenge.
+ * @param {BigInt} previousAssertion - The previous assertion id.
+ * @returns {Object} - An object containing the nodes, assertions, confirmData, and confirmHash.
+ */
+export const submitMockRollupChallenge = async (
+    referee, 
+    challengerWallet, 
+    mockRollup, 
+    refereeCalculations, 
+    currentAssertion,
+    previousAssertion
+) => {
+    
+    //set isCheckingAssertions to true in referee
+    let isAssertionChecking = await referee.isCheckingAssertions();
+    if (!isAssertionChecking) {
+        await referee.toggleAssertionChecking();
+    }
+    
+    //set referee rollup address to mock rollup
+    const prevRollupAddress = await referee.rollupAddress();
+    const mockRollupAddress = await mockRollup.getAddress();
+    await referee.setRollupAddress(mockRollupAddress);
+    
+    // Create a Node on the MockRollup
+    let nodeArray = [];
+    let assertionArray = [];
+    let lastNodeBlockNumber = 0;
+    for (let i = previousAssertion + 1; i <= currentAssertion; i++) {
+        //create new node on mock rollup
+        let hexStr = "0x" + BigInt(i).toString(16).padStart(64, "0"); //make unique hex string
+        const trx = await mockRollup.createNode(
+            i,
+            hexStr,
+            hexStr
+        );
+        
+        //read the Node from the MockRollup to get Node data
+        const node = await mockRollup.getNode(i);
+        
+        //push node into array
+        nodeArray.push(node);
+        assertionArray.push(i);
+        
+        //update last block number
+        lastNodeBlockNumber = node.createdAtBlock;
+    }
+    
+    //get confirm data and hash from contract
+    const [confirmData, confirmHash] = await refereeCalculations.getConfirmDataMultipleAssertions(
+        assertionArray, 
+        mockRollupAddress
+    );
+    
+    // Submit a challenge pointing to the MockRollup Node(s)
+    await referee.connect(challengerWallet).submitChallenge(
+        currentAssertion,
+        previousAssertion,
+        confirmHash, 
```

Comment:
I believe this should only be the confirmHash if it is a batch(multi-assertion challenge).

For non-batch challenges, I may be incorrect, but I believe this should only be the confirm data from the assertion, not a hash of the confirm data.